lib/src/barback/asset_environment.dart

```diff
@@ -447,25 +475,15 @@ class AssetEnvironment {
 
   /// Pauses sending source asset updates to barback.
   void pauseUpdates() {
-    // Cannot pause while already paused.
-    assert(_modifiedSources == null);
-
-    _modifiedSources = new Set<AssetId>();
+    assert(_paused == false);
+    _paused = true;
```

Comment:
This indirection to a setter with a lot of side-effects seems pretty confusing to me. In general, I don't expect a setter to do a lot. I'd probably keep the behavior in these methods, and then add an `_isPaused` getter to access the state.